DefaultSqlQueryBuilder/Extensions/SqlClauseExtensions.cs

```diff
@@ -63,10 +63,11 @@ public static IEnumerable<SqlClause> ConsolidateWhereClauses(this IEnumerable<Sq
 				.Cast<WhereSqlClause>()
 				.ToArray();
 
-			if (removableWhereClauses.Any())
+			if (removableWhereClauses.Count() > 1)
 			{
+				int lastWhereClause = clauses.FindLastIndex(clause => clause is WhereSqlClause);
+				clauses[lastWhereClause] = removableWhereClauses.Aggregate(MergeWhere);
 				clauses.RemoveAll(removableWhereClauses.Contains);
```

Comment:
will this .RemoveAll() remove the aggregate we just added in the line above?

Reply:
It will not remove it since it only removes objects that are also in the `removableWhereClauses` collection. The aggregate is a new and unique object (except when there is only one WHERE clause, in which case we skip this step entirely).
I should note that an internal fork of this project is being used in my Vega IT internship project which means changes are being tested 😄

Reply:
you're completely right :) thanks for this PR! :) I'll merge it now :)